Sources/SwiftFormat/Rules/NoEmptyLineOpeningClosingBraces.swift

```diff
@@ -0,0 +1,130 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+@_spi(Rules)
+public final class NoEmptyLinesOpeningClosingBraces: SyntaxFormatRule {
+  public override func visit(_ node: AccessorBlockSyntax) -> AccessorBlockSyntax {
+    var result = node
+    switch node.accessors {
+      case .accessors(let accessors):
+        result.accessors = .init(rewritten(accessors))
+      case .getter(let getter):
+        result.accessors = .init(rewritten(getter))
+    }
+    result.rightBrace = rewritten(node.rightBrace)
+    return result
+  }
+  
+  public override func visit(_ node: CodeBlockSyntax) -> CodeBlockSyntax {
+    var result = node
+    result.statements = rewritten(node.statements)
+    result.rightBrace = rewritten(node.rightBrace)
+    return result
+  }
+  
+  public override func visit(_ node: MemberBlockSyntax) -> MemberBlockSyntax {
+    var result = node
+    result.members = rewritten(node.members)
+    result.rightBrace = rewritten(node.rightBrace)
+    return result
+  }
+  
+  public override func visit(_ node: ClosureExprSyntax) -> ExprSyntax {
+    var result = node
+    result.statements = rewritten(node.statements)
+    result.rightBrace = rewritten(node.rightBrace)
+    return ExprSyntax(result)
+  }
+  
+  func rewritten(_ token: TokenSyntax) -> TokenSyntax {
+    let (trimmedLeadingTrivia, count) = token.leadingTrivia._trimmingSuperfluousNewlines()
+    if trimmedLeadingTrivia.sourceLength != token.leadingTriviaLength {
+      diagnose(.removeEmptyLinesBefore(count), on: token, anchor: .start)
+      return token.with(\.leadingTrivia, trimmedLeadingTrivia)
+    } else {
+      return token
+    }
+  }
+  
+  func rewritten<C: SyntaxCollection>(_ collection: C) -> C {
+    var result = collection
+    if let first = collection.first, first.leadingTrivia.containsNewlines,
+       let index = collection.index(of: first)
+    {
+      let (trimmedLeadingTrivia, count) = first.leadingTrivia._trimmingSuperfluousNewlines()
+      if trimmedLeadingTrivia.sourceLength != first.leadingTriviaLength {
+        diagnose(.removeEmptyLinesAfter(count), on: first, anchor: .leadingTrivia(0))
+        result[index] = first.with(\.leadingTrivia, trimmedLeadingTrivia)
+      }
+    }
+    return rewrite(result).as(C.self)!
+  }
+}
+
+extension Trivia {
+  func trimmingSuperfluousNewlines() -> Trivia {
```

Comment:
It looks like this function is never used, only the one with the underscore below is. Delete this and remove the underscore from the other one?

Reply:
Yeah, I forget this 😅